wcfsetup/install/files/lib/system/language/preload/command/CachePreloadPhrases.class.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace wcf\system\language\preload\command;
+
+use Laminas\Diactoros\Stream;
+use wcf\data\language\Language;
+use wcf\system\event\EventHandler;
+use wcf\system\language\preload\event\PreloadPhrasesCollecting;
+use wcf\util\StringUtil;
+
+/**
+ * Rebuilds the phrase preload cache for the
+ * requested language.
+ *
+ * @author Alexander Ebert
+ * @copyright 2001-2022 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System\Language\Preload\Command
+ * @since 6.0
+ */
+final class CachePreloadPhrases
+{
+    private readonly EventHandler $eventHandler;
+    private readonly Language $language;
+
+    public function __construct(Language $language)
+    {
+        $this->language = $language;
+
+        $this->eventHandler = EventHandler::getInstance();
+    }
+
+    public function __invoke(): void
+    {
+        $event = new PreloadPhrasesCollecting($this->language);
+        $this->eventHandler->fire($event);
+
+        $file = new Stream(\WCF_DIR . $this->language->getPreloadCacheFilename(), 'wb');
```

Comment:
We likely need the `AtomicWriter`, instead of a Laminas Stream here. Otherwise multiple requests might write into the file at the same time. The method calls should likely be identical, it might require an additional `->flush()` before the `->close()`, though.